components/ws-manager/pkg/manager/monitor.go

```diff
@@ -380,9 +380,27 @@ func actOnPodEvent(ctx context.Context, m actingManager, status *api.WorkspaceSt
 		_, alreadyFinalized := wso.Pod.Annotations[startedDisposalAnnotation]
 
 		if (terminated || gone) && !alreadyFinalized {
-			// We start finalizing the workspace content only after the container is gone. This way we ensure there's
-			// no process modifying the workspace content as we create the backup.
-			go m.finalizeWorkspaceContent(ctx, wso)
+			if isFailed, ok := wso.Pod.Annotations[workspaceFailedBeforeStoppingAnnotation]; ok && isFailed == "true" {
```

Comment:
```suggestion
			if wso.Pod.Annotations[workspaceFailedBeforeStoppingAnnotation]== api.WorkspaceConditionBool_TRUE {
```

Reply:
It appears that `api.WorkspaceCondtionBool_TRUE` is not being used as a pod annotation.
https://github.com/gitpod-io/gitpod/blob/aec1c9c360fbfe0c1e267b4eef2e4ac42afe38b8/components/ws-manager/pkg/manager/status.go#L319-L320

https://github.com/gitpod-io/gitpod/blob/aec1c9c360fbfe0c1e267b4eef2e4ac42afe38b8/components/ws-manager/pkg/manager/monitor.go#L271

Reply:
I added the bool value for the annotations.